bookkeeper-http/twitter-http-server/src/main/java/org/apache/bookkeeper/http/twitter/TwitterHttpHandlerFactory.java

```diff
@@ -0,0 +1,268 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http.twitter;
+
+import com.twitter.finagle.http.Request;
+import com.twitter.finagle.http.Response;
+import com.twitter.util.Future;
+
+import org.apache.bookkeeper.http.AbstractHttpHandlerFactory;
+import org.apache.bookkeeper.http.HttpServiceProvider;
+
+
+
+/**
+ * Factory which provide http handlers for TwitterServer based Http Server.
+ */
+public class TwitterHttpHandlerFactory extends AbstractHttpHandlerFactory<TwitterAbstractHandler> {
```

Comment:
Maybe it would be better that this 'Twitter' custom http server would be packaged only inside a Twitter project and not inside the community edition ? What is the reason to have this implementation in Apache codebase ?

Anyway I am fine with keeping it here if there is a reason, it is an additional dependency and clients are not required to import it


//cc @sijie @jvrao

Reply:
@eolivelli the idea is similar as stats provider. so not everyone will import this dependency, unless you configure to use this http server implementation.

Reply:
Fine for me

Reply:
This follows @yzang's change in #278 . it maybe make a more easy usage for twitter. and also provide another example of how to implement the factory besides the Vertx one. But indeed it introduced "twitter-server". 
also @yzang to comments on this.